client/allocrunner/taskrunner/artifact_hook_test.go

```diff
@@ -158,3 +159,230 @@ func TestTaskRunner_ArtifactHook_PartialDone(t *testing.T) {
 	require.True(t, resp.Done)
 	require.Len(t, resp.State, 2)
 }
+
+// TestTaskRunner_ArtifactHook_ConcurrentDownload asserts that the artifact hook
+// download multiple files concurrently. this is a successful test without any errors.
+func TestTaskRunner_ArtifactHook_ConcurrentDownloadSuccess(t *testing.T) {
+	t.Parallel()
+
+	me := &mockEmitter{}
+	artifactHook := newArtifactHook(me, testlog.HCLogger(t))
+
+	// Create a source directory all 7 artifacts
+	srcdir, err := ioutil.TempDir("", "nomadtest-src")
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, os.RemoveAll(srcdir))
+	}()
```

Comment:
You can shorten this code to the following (we just haven't fixed up all of our old tests yet):

```suggestion
	srcdir := t.TempDir()
```

Reply:
resolved via 1ff65dc2f

Reply:
resolved via 1ff65dc2f